src/util/parse_options.h

```diff
@@ -35,4 +36,23 @@ class parse_options_baset
   bool parse_result;
 };
 
+class parse_optionst: public parse_options_baset
```

Comment:
I think we could remove `parse_optionst` and include the new code directly in `parse_options_baset`.